README.md

```diff
@@ -2,16 +2,82 @@
 
 # CSI attacher
 
-The csi-attacher is part of Kubernetes implementation of [Container Storage Interface (CSI)](https://github.com/container-storage-interface/spec).
+The external-attacher is a sidecar container that attaches volumes to nodes by calling `ControllerPublish` and `ControlerUnpublish` functions of CSI drivers. It is necessary because internal Attach/Detach controller running in Kubernetes controller-manager does not have any direct interfaces to CSI drivers.
+
+## Terminology
+
+In Kubernetes, term *attach* means 3rd party volume attachment to a node. This is common in cloud environments, where cloud API is able to attach a volume to a node without any code running on the node. In CSI terminology, this corresponds to `ControllerPublish` call.
+
+*Detach* then means reverse operation, 3rd party volume detachment from a node, `ControllerUnpublish` in CSI terminology.
+
+It is **not** attach/detach operation performed by a code running on a node, such as attachment of iSCSI or Fibre Channel volumes. These are typically performed during `NodeStage` and `NodeUnstage` CSI calls and are not done by the external-attacher.
 
 ## Overview
+The external-attacher is an external controller that monitors `VolumeAttachment` objects created by controller-manager and attaches/detaches volumes to/from nodes (i.e. calls `ControllerPublish`/`ControllerUnpublish`. Full design can be found at Kubernetes proposal at [container-storage-interface.md](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/container-storage-interface.md)
+
+## Usage
```

Comment:
Add a compatibility matrix here, like how node-driver-registrar does

Reply:
Added, although I am not sure I understand the matrix correctly.